llama_index/tools/tool_spec/load_and_search/base.py

```diff
@@ -0,0 +1,146 @@
+
+        You are required to pass the natural language query argument when calling this endpoint
+
+        Args:
+            query (str): The natural language query used to retreieve information from the index
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        tool: FunctionTool,
+        index_cls: Type[BaseIndex],
+        index_kwargs: Dict,
+        metadata: ToolMetadata,
+        index: Optional[BaseIndex] = None,
+    ) -> None:
+        """Init params."""
+        self._index_cls = index_cls
+        self._index_kwargs = index_kwargs
+        self._index = index
+        self._metadata = metadata
+        self._tool = tool
+
+        if self._metadata.name is None:
+            raise ValueError("Tool name cannot be None")
+        self.spec_functions = [
+            self._metadata.name,
+            "read_{}".format(self._metadata.name),
+        ]
+        self._tool_list = [
+            FunctionTool.from_defaults(
+                fn=self.load,
+                name=self._metadata.name,
+                description=self.loader_prompt.format(
+                    self._metadata.name, self._metadata.description
+                ),
+                fn_schema=self._metadata.fn_schema,
+            ),
+            FunctionTool.from_defaults(
+                fn=self.read,
+                name=str("read_{}".format(self._metadata.name)),
+                description=self.reader_prompt.format(metadata.name),
+                fn_schema=create_schema_from_function("ReadData", self.read),
+            ),
+        ]
+
+    @property
+    def metadata(self) -> ToolMetadata:
+        return self._metadata
+
+    @classmethod
+    def from_defaults(
+        cls,
+        tool: FunctionTool,
+        index_cls: Optional[Type[BaseIndex]] = None,
+        index_kwargs: Optional[Dict] = None,
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+        fn_schema: Optional[Type[BaseModel]] = None,
+    ) -> "LoadAndSearchToolSpec":
+        """From defaults."""
+
+        index_cls = index_cls or VectorStoreIndex
+        index_kwargs = index_kwargs or {}
+        if name is None:
+            name = tool.metadata.name
+        if description is None:
+            description = tool.metadata.description
+        if fn_schema is None:
+            fn_schema = tool.metadata.fn_schema
+        metadata = ToolMetadata(name=name, description=description, fn_schema=fn_schema)
+        return cls(
+            tool=tool,
+            index_cls=index_cls,
+            index_kwargs=index_kwargs,
+            metadata=metadata,
+        )
+
+    def to_tool_list(
+        self,
+        func_to_metadata_mapping: Optional[Dict[str, ToolMetadata]] = None,
+    ) -> List[FunctionTool]:
+        return self._tool_list
+
+    def load(self, *args: Any, **kwargs: Any) -> Any:
+        # Call the wrapped tool and save the result in the index
+        docs = self._tool(*args, **kwargs).raw_output
```

Comment:
@logan-markewich @ajhofmann i didn't test this at all but yolo

Reply:
you can also call `str(self._tool(...))` if you just want the string